Add file with hw_3

MastenaM opened the pull request.

MastenaM committed:
Add file with hw_3

pacifikus reviewed (changes requested):
Добрый день!

Спасибо за работу! В целом все хорошо, оставлю несколько комментариев о том, что можно улучшить:

- Процент выживаемости в каждом классе можно посчитать чуть более лаконично:
```
df.groupby('Pclass')['Survived'].mean() * 100
```
- Немного странно у женщин видеть имя "William". Видимо, это имя мужа, которое являлось составной частью полного имени замужней женщины в Англии, а алгоритм получения имени (first name) самой женщины может отличаться от алгоритма для мужских имен... Стоит попробовать выделить извлечение имени в отдельную функцию, которая учитывает, например, есть ли в имени женщины приставка "Mrs", указывающая на замужество, и соответствующим образом после обрабатывает имя.
- Количества n-местных кабин (в которых было 2, 3, 4, ... человека) также можно посчитать чуть проще, комбинирую группировку и value_counts:
```
df.groupby('Cabin').size().value_counts()[1:]
```

MastenaM committed:
Переделала задание про топ женских имен.

Исправила ошибку, когда для замужних дам в столбцах с именами были имена мужей, а не их. Теперь топ считается только по именам женщин.

pacifikus reviewed (approved):
Добрый день!

Спасибо за работу! Все отлично!
